web/js/codeworld.js

```diff
@@ -35,22 +35,19 @@ async function init() {
     preloadBaseBundle();
     window.setInterval(preloadBaseBundle, 1000 * 60 * 60);
 
-    window.allProjectNames = [
-        []
-    ];
-    window.allFolderNames = [
-        []
-    ];
+    initDirectoryTree();
+
     window.openProjectName = null;
-    window.nestedDirs = [''];
 
     window.savedGeneration = null;
     window.runningGeneration = null;
 
     if (window.location.pathname === '/haskell') {
         window.buildMode = 'haskell';
+        window.projectEnv = 'haskell';
```

Comment:
Can you explain why you need this to be different from window.buildMode?

Reply:
Funblocks build using `codeworld` mode, but sources are in `data/blocklyXML/projects` folder. So I used different variable to separate build mode and place from which need to load projects between `codeworld.js` and `funblocks.js`.